Add logging of service metadata during instance create/update/delete

kathap opened the pull request:
Inform operators which service plan, service offering and broker is involved in service instance creation/update/deletion

* A short explanation of the proposed change:
Add names of service plan, service offering and broker as logging information for service instance creation/update/deletion.
This change updates fetch_writable_service_instance to eagerly load associated service_plan, service, and service_broker relationships for managed service instances. This avoids N+1 queries when accessing these associations later in the request lifecycle.

* An explanation of the use cases your change solves
This change improves operator visibility by logging service plan names, offerings, and broker names during service instance creation, update, and deletion.
Previously, logs only included GUIDs or IDs, requiring manual lookups or customer input to identify the service involved. With this change, operators can directly see which service is affected, making it easier to diagnose issues and support users/customers more efficiently.

* [X] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/CONTRIBUTING.md)

* [X] I have viewed, signed, and submitted the Contributor License Agreement

* [X] I have made this pull request to the `main` branch

* [X] I have run all the unit tests using `bundle exec rake`

* [X] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/spec/README.md#cf-acceptance-tests-cats)

kathap committed:
Inform operators which service plan, offering and broker is involved in service instance creation/update/deletion

kathap committed:
Refactor service plan lookup to eager load service and broker associations

kathap committed:
simplify update managed with adding only 1 simple sql query

kathap committed:
use more specific joins for delete request

kathap committed:
Add tests for the log info

svkrieger committed:
Eager load associations of service instance

philippthun commented on app/controllers/v3/service_instances_controller.rb:
```diff
@@ -297,6 +311,13 @@ def update_managed(service_instance)
+        "Updating managed service instance with name '#{service_instance.name}' " \
+        "using service plan '#{service_instance.service_plan.name}' " \
+        "from service offering '#{service_instance.service_plan.service.label}' " \
+        "provided by broker '#{service_instance.service_plan.service.service_broker.name}'."
```
When I tested this, a database query to fetch broker information was executed.

It seems as if `service_instance.reload` ([here](https://github.com/cloudfoundry/cloud_controller_ng/blob/511e64b9d933de1fe9454e2ccc5108e4a0e9b6a7/app/actions/v3/service_instance_update_managed.rb#L272)) removes the associated objects (or at least some of them).

So for the `update_managed` part, eager loading does not have the expected benefits.

Additional question: can the service plan be updated? This would even make the eager loading completely obsolete in such a case.

svkrieger committed:
Remove unnecessary reload of service_instance

Reloading the service_instance just ensures that follow-up preflight checks run with the original service instance values.
This can also be achieved by applying changes on a copy of the object. With that, we don't loose eagerly loaded associations and save a few DB queries.

svkrieger committed:
Further reduce DB queries
